src/conways.rs

```diff
@@ -17,10 +19,176 @@ impl ConwaysGrid {
         Self { grid }
     }
 
+    fn modify_cell(&mut self, pos: Position, new_state: CellState) {
+        if let Some(grid_row) = self.grid.get_mut(pos.0) {
+            if let Some(cell) = grid_row.get_mut(pos.1) {
+                *cell = new_state;
+            }
+        }
+    }
+
+    pub fn from_alive_cells(alive_cells: Vec<Position>) -> Self {
```

Comment:
I think it's better to pass as a reference here
`pub fn from_alive_cells(alive_cells: &[Position])`